### ISSUE TITLE
Cache tags based revalidation

### ISSUE THREAD

**Issue by lauriii** [chapter-three/next-drupal#235]
The current revalidation logic in _next_extras_ has some potential scalability issues since it's using paths for revalidating pages. It's complex to use this approach with lists and entity references since currently those would have to be added manually to the revalidation logic. If we implemented that while using the path based approach, with larger sites, this could lead into hundreds or thousands of requests. 

Since Drupal provides cache tags for all requests, I'm wondering if we could utilize cache tags for the revalidation.

**Comment by shadcn** [comment]
Adding links to some ongoing discussions:

https://github.com/vercel/next.js/discussions/34585
https://github.com/vercel/next.js/discussions/34567

**Comment by hanoii** [comment]
After a few days of installing next.js + Drupal and getting familiar with most things this was the first major thing I was curious to see if it was planned, or something was on the line. I see there's a potential limitation on next.js , right?

Anyway, I want to bring in something I used for purging CF without having enterprise (which is where tag invalidation is present): https://www.drupal.org/project/purge_queuer_url

Maybe a purge plugin can be implemented for the revalidation paths or use the module/logic some other way.

**Comment by shadcn** [comment]
@hanoii Yep. Still not available. But it's being worked on. For now, what I'm doing is using a custom `Revalidator` plugin that reads the entity tags and then revalidate the paths (this assumes that the Next.js paths to revalidate are deterministic).

**Comment by hanoii** [comment]
Would you be able to share that custom revalidator of yours? How do you translate tags to paths?

**Comment by hanoii** [comment]
@shadcn any chance you can share that custom revalidator? Thanks!

**Comment by hanoii** [comment]
Still trying to see if this progressed a bit, saw 

https://twitter.com/shadcn/status/1629533541183651842
And 
https://vercel.com/blog/vercel-cache-api-nextjs-cache

Is this going to be a vercel only thing?

Any chance you can share your custom revalidation work?
